src/components/ClientSDKProvider/ClientSDKProvider.tsx

```diff
@@ -13,6 +15,10 @@ export const ClientSDKProvider: React.FC<ClientSDKProviderProps> = ({
   children,
 }) => (
   <ClientSDKContext.Provider value={client}>
-    <ClientSDKCacheProvider client={client}>{children}</ClientSDKCacheProvider>
+    <ClientSDKCacheProvider client={client}>
+      <ClientSDKProxiedContext.Provider value={wrapInProxy(client)}>
```

Comment:
Can we have it as a separate component (like `ClientSDKCacheProvider`) to apply all proxy logic there?